File: api/src/main/java/com/example/api/services/DishService.java
package com.example.api.services;

import java.util.List;

import com.example.api.models.Dish;
import com.example.api.models.Restaurant;

public interface DishService {
    Dish createDish(Dish dish);

    List<Dish> getRestorerDishsById(Restaurant restaurant);
}

File: client/src/app/restaurant/dish/dish.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiResponse } from 'src/app/interfaces/ApiResponseInterface';
import { Dish } from 'src/app/interfaces/DishInterface';
import { ApiService } from 'src/app/services/api/api.service';

@Injectable({
    providedIn: 'root',
})
export class DishService {
    constructor(private apiService: ApiService) {}
    apiUrl: string = '/dish';

    createDish(data: Dish): Observable<ApiResponse<Dish>> {
        return this.apiService.post<ApiResponse<Dish>, Dish>(this.apiUrl, data);
    }

    getDishs(): Observable<ApiResponse<Dish[]>> {
        return this.apiService.get<ApiResponse<Dish[]>>(`${this.apiUrl}/1`);
    }
}
